### DISCUSSION TITLE
How retriving a simple list of field names ?

### DISCUSSION THREAD

**Discussion by jossefaz** [pydantic/pydantic#3697] [Question] [answered]
I have a class

```python
    class A(BaseModel):
        a:int
        b:str
```
I would like to simply retrieve a list of field names :  `['a','b']`
So far I retrieved them by using this

`set(A.__fields__.keys())`

Did I miss any API method dedicated to this purpose or is it the correct way to do that ? I didn't found in the documentation

**Comment by uriyyo** [comment, marked as answer]
That's correct. There is no method to do that.

Btw, in my code when I need a list of model fields I use the code below:
```py
names = [*A.__fields__]
```
